.github/workflows/startproject.yml

```
@@ -0,0 +1,88 @@
+name: Start Project
+
+on:
+    pull_request:
+    push:
+      branches: [main]
+
+# Restrict jobs in this workflow to have no permissions by default; permissions
+# should be granted per job as needed using a dedicated `permissions` block
+permissions: {}
+
+jobs:
+  creates-valid-project:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+
+    services:
+      postgres:
+        image: postgres:14
+        env:
+          POSTGRES_USER: dev
+          POSTGRES_PASSWORD: dev_password
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+            # Allows us to use the same env config used locally by mapping to port to
+            # the test runner container.
+            - 5432:5432
+
+    steps:
+      - uses: actions/checkout@v4
```

suggestion: we should explicitly disable persisted credentials since they're not needed for this workflow

```suggestion
      - uses: actions/checkout@v4
        with:
          persist-credentials: false
```